src/discovery/discovery_options.rs

```diff
@@ -0,0 +1,31 @@
+use derive_builder::Builder;
+
+/// Factories discovery options
+#[derive(Debug, Builder, Default)]
+pub struct FactoriesDiscoveryOptions {
+    /// From block number, if None then the discovery start block number will be 10000835 (the creation block of UniswapV2Factory)
+    #[builder(default = "10000835")]
+    pub from_block: u64,
+    /// To block number, if None then the discovery end block number will be the current block number
+    #[builder(default)]
+    pub to_block: Option<u64>,
+    /// Block number step
+    #[builder(default = "1000")]
+    pub step: u64,
+    /// Filter factory that have to has at least a number of pairs
+    pub number_of_amms_threshold: u64,
+}
+
+/// ERC4626 vault discovery options
+#[derive(Debug, Builder, Default)]
+pub struct Erc4626DiscoveryOptions {
+    /// From block number, if None then the discovery start block number will be 0
+    #[builder(default = "0")]
+    pub from_block: u64,
+    /// To block number, if None then the discovery end block number will be the current block number
+    #[builder(default)]
+    pub to_block: Option<u64>,
+    /// Block number step
+    #[builder(default = "1000")]
+    pub step: u64,
+}
```

Comment:
Can we move each of these to the file that they correlate with instead of creating a new file for these types?

Reply:
sure